include/clap/ext/draft/preset-load.h

```diff
@@ -9,19 +9,13 @@ extern "C" {
 #endif
 
 typedef struct clap_plugin_preset_load {
-   // Loads a preset in the plugin native preset file format from a path.
-   // [main-thread]
-   bool(CLAP_ABI *from_file)(const clap_plugin_t *plugin, const char *path);
-
-   // Loads a preset in the plugin native preset file format from a path which points to a preset
-   // container file.
-   // preset_id must be a valid string, which is specific to the plugin itself and identifies the
-   // desired preset within the given preset container.
+   // Loads a preset in the plugin native preset file format from a URI. eg:
+   // - "file:///home/abique/.u-he/Diva/Presets/Diva/HS Bass Nine.h2p"
+   // - "file:///home/abique/my-sound-bank/<preset-id>"
+   // - "plugin://<plugin-id>/<preset-id>"
    //
    // [main-thread]
-   bool(CLAP_ABI *from_container_file)(const clap_plugin_t *plugin,
-                                       const char          *path,
-                                       const char          *preset_id);
+   bool(CLAP_ABI *from_uri)(const clap_plugin_t *plugin, const char *uri);
```

Comment:
What if a plugin doesn't have a way to pop up a GUI file dialog while its GUI isn't visible? Or should the plugin always create a standalone window in that case?

Reply:
Good point. Then I think as you said before, it should always be the host's responsibility. I suggest adding a comment saying something like `// If the preset cannot be loaded for any reason, return false so that the host can process the error as it likes. The plugin should not present error UI/dialogs itself.`

Reply:
It might be nice to have some way for the plugin to return a one line error string or something though, since there are a number of potential causes for why the preset could not be loaded.

Reply:
Maybe the plugin can use host_log?
Then the host can simply do:
```C
x = get_last_error_msg_index();
if (!plug->load_from_uri()) {
   y = get_last_error_msg_index();
   // report all the errors from x to y into the error message
}
```

Otherwise the issue is that the host has to pre-allocate the error buffer, which makes the interface a bit more bloated.

Reply:
If the host tries to `use load_from_uri()` and it fails, then it can simply show: "failed to load preset <uri>".
That should be enough for the user, and then he can investigate the engine logs.

Reply:
> Otherwise the issue is that the host has to pre-allocate the error buffer, which makes the interface a bit more bloated.

It can be allocated on the stack, so that shouldn't make the API much more cumbersome to use. The error buffer could either have a predefined size, or the function could get a `char*` and a `size_t`/`uint32_t` parameter for the buffer and the buffer's size,and a `size_t*`/`uint32_t*` where the plugin can change the error length to. If that doesn't change from the 0 value the host initialized it to, then there was no error message.